src/Microsoft.Graph.Core/Requests/BaseFunctionMethodRequestBuilder.cs

```diff
@@ -48,25 +44,21 @@ public abstract class BaseFunctionMethodRequestBuilder<T> : BaseRequestBuilder w
         /// <returns>The built request.</returns>
         public T Request(IEnumerable<Option> options = null)
         {
-            string fnUrl = this.RequestUrl;
-
-            if (this.passParametersInQueryString && this._queryOptions.Count > 0)
+            // Validate the query options according to 
+            // http://docs.oasis-open.org/odata/odata/v4.01/odata-v4.01-part1-protocol.html#sec_InvokingaFunction
+            if (options != null)
             {
-                if (options == null)
-                {
-                    options = this._queryOptions;
-                }
-                else
+                var unexpectedQueryOptions = options.OfType<QueryOption>().Any<QueryOption>(qp => 
+                                                                                 qp.Name.ToLower() != "filter" && 
+                                                                                 qp.Name.ToLower() != "orderby");
+
+                if (unexpectedQueryOptions)
                 {
-                    options.ToList().AddRange(this._queryOptions);
+                    throw new ArgumentException("You can only use filter and orderby query options with this function.");
```

Comment:
1. Is this a breaking change? Are all query options being accepted before this change goes in? 
2. It would be helpful if this is stated in the summary of the function.

Reply:
I think this is moving the error from service side to client side, i.e. an early failure in the stack. Technically breaking, but I wouldn't count it as something that requires a major version bump.

Reply:
1) I haven't seen functions used with query params. With that stated, we arent a strict OData service so doing this may not be the best thing todo.  I'm going to remove this check.
2) Will do. Note that this is only (so far) been called from generated classes.

Reply:
I removed this check. 1) I can't totally trust that I know whether these querystrings aren't in actual use, 2) and into the future, I don't totally trust the service to not support these.